by-language/python-sqlalchemy/insert_pandas.py

```diff
@@ -0,0 +1,155 @@
+"""
+About
+=====
+
+Example program to demonstrate efficient batched inserts using CrateDB and
+pandas, based on SQLAlchemy's `insertmanyvalues` vs. CrateDB's bulk import
+HTTP endpoint.
+
+- https://docs.sqlalchemy.org/core/connections.html#controlling-the-batch-size
+- https://crate.io/docs/crate/reference/en/latest/interfaces/http.html#bulk-operations
+
+
+Setup
+=====
+::
+
+    pip install --upgrade click colorlog 'crate[sqlalchemy]' pandas sqlalchemy
+
+
+Synopsis
+========
+::
+
+    # Run CrateDB.
+    docker run --rm -it --publish=4200:4200 crate
+
+    # Use local CrateDB.
+    time python insert_pandas.py
+
+    # Use local CrateDB with "basic" mode.
+    time python insert_pandas.py --mode=basic --insertmanyvalues-page-size=5000
+
+    # Use local CrateDB with "bulk" mode, and a few more records.
+    time python insert_pandas.py --mode=bulk --bulk-size=20000 --num-records=75000
+
+    # Use CrateDB Cloud.
+    time python insert_pandas.py --dburi='crate://admin:<PASSWORD>@example.aks1.westeurope.azure.cratedb.net:4200?ssl=true'
+
+
+Details
+=======
+To watch the HTTP traffic to your local CrateDB instance, invoke::
+
+    sudo ngrep -d lo0 -Wbyline port 4200
+
+"""
+import logging
+
+import click
+import colorlog
+import pkg_resources
+import sqlalchemy as sa
+from colorlog.escape_codes import escape_codes
+from pandas._testing import makeTimeDataFrame
+
+from crate.client.sqlalchemy.support import insert_bulk
+
+logger = logging.getLogger(__name__)
+
+pkg_resources.require("sqlalchemy>=2.0")
```

Comment:
Also reflect this within the `requirements.txt` file, and probably also within `insert_dask.py`.

Reply:
Fixed.

Reply:
Is it still needed at all then?